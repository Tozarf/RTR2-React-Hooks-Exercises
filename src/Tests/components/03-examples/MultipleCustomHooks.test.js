import { shallow } from 'enzyme'
import React from 'react'
import { MultipleCustomHooks } from "../../../components/03-examples/MultipleCustomHooks"

import { useCounter } from '../../../hooks/useCounter'
import { useFetch } from '../../../hooks/useFetch'
jest.mock('../../../hooks/useFetch')
jest.mock('../../../hooks/useCounter') //This is how we coul mock hooks to be used below for testing



describe('Tesing <MultipleCustomHooks/>', () => {
    
    beforeEach( () => {
        useCounter.mockReturnValue({
            counter: 10,
            increment: () => {}
        })
    })

    test('should match with snapshot', () => {
        
        useFetch.mockReturnValue({
            data:null,
            loading:true,
            error: null
        })
        
        const wrapper = shallow(<MultipleCustomHooks/>)

        expect (wrapper).toMatchSnapshot();
    })
    
    test('should show information', () => {
        useFetch.mockReturnValue({
            data:[{
                author : "Fausto",
                quote : "tired af"

            }],
            loading:false,
            error: null
        })
        const wrapper = shallow(<MultipleCustomHooks/>)

        expect( wrapper.find(".alert").exists()).toBe( false )
        expect( wrapper.find(".mb-0").text().trim()).toBe( "tired af")
        expect( wrapper.find("footer").text().trim()).toBe( "Fausto")


    })
    
    
})
