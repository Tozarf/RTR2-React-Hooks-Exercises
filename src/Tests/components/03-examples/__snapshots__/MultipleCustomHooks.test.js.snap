// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tesing <MultipleCustomHooks/> should match with snapshot 1`] = `
<div>
  <h1>
    Breaking Bad Quotes
  </h1>
  <hr />
  <div
    className="alert alert-info text-center"
  >
    Loading...
  </div>
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Next Quote
  </button>
</div>
`;
